### PR TITLE
return device classes as an option in user.getMyDevices, this enables…

### PR DESCRIPTION
**Pull request by SmbatYeranyan**
… easy access to Model class and device operations

### CODE REVIEW

File: src/entities/User.js [hunk 1]
```diff
@@ -1,64 +1,74 @@
...
+    getMyDevices(opts = {}) {
+        return new Promise((resolve, reject) => {
+            this.getUserInfo().then(() => {
+
+                this.ajax.get(`/users/${this.userInfo.id}/devices`).then((response) => {
 
-      }
-    });
-  }
+                    if (opts.asClasses) {
```
Comment (jontore) [inline src/entities/User.js]: I think a good refactoring in the future is to have a super entity with helpers for these things that all the entities can inherit from.

File: test/user.test.js [hunk 2]
```diff
@@ -56,37 +58,63 @@ describe('User', function() {
...
+
+        it('should get all devices', function(done) {
+            userInstance.userInfo = userStub;
+            userInstance.getMyDevices().then((devices) => {
+                expect(devices).to.deep.equal(devicesStub);
+                done();
+            });
+
+            setTimeout(() => {
+                //give some time for the async userInfo cache to take effect
```
Comment (jontore) [inline test/user.test.js]: why 100, if you want it to be async 1 should be enough. And we're not slowing down the test suite.

Comment (jontore) [inline test/user.test.js]: I also don't understand why it is necessary.

File: test/user.test.js [hunk 3]
```diff
@@ -56,37 +58,63 @@ describe('User', function() {
...
+                this.requests[0].respond(200, {
+                    'Content-Type': 'text/json'
+                }, JSON.stringify(devicesStub));
+            }, 100);
+        });
+
+    });
```
Comment (jontore) [inline test/user.test.js]: What happen when this fails.

**Comment by jontore** [comment]
Good idea, maybe something we should do consistently with other methods as well.

File: src/entities/User.js (hunk 1 above)
Comment (SmbatYeranyan) [inline src/entities/User.js]: agreed, this was a trial run to see how we can use it

**Pull request merged**